lib/class/api.class.php

```diff
@@ -251,25 +264,49 @@ public static function handshake($input)
                 $db_results = Dba::read($sql);
                 $catalog    = Dba::fetch_assoc($db_results);
 
-                echo XML_Data::keyed_array(array('auth'=>$token,
-                    'api'=>self::$version,
-                    'session_expire'=>date("c",time()+AmpConfig::get('session_length')-60),
-                    'update'=>date("c",$row['update']),
-                    'add'=>date("c",$row['add']),
-                    'clean'=>date("c",$row['clean']),
-                    'songs'=>$song['song'],
-                    'albums'=>$album['album'],
-                    'artists'=>$artist['artist'],
-                    'playlists'=>$playlist['playlist'],
-                    'videos'=>$vcounts['video'],
-                    'catalogs'=>$catalog['catalog']));
+                if ($outputFormat == 'json') {
+                  echo json_encode(
+                      array(
```

Comment:
I think you should avoid repeating the array, which is really error prone when editing the array (and then not replicating it in every branch of the if).

Why not using an auxiliary variable to store the resulting array, and just call the right formatting method on it?